test/e2e/parallel/nstemplatetier_test.go

```diff
@@ -184,6 +184,12 @@ func setupSpaces(t *testing.T, awaitilities wait.Awaitilities, tier *tiers.Custo
 		name := fmt.Sprintf(nameFmt, i)
 		s, _, _ := CreateSpace(t, awaitilities, testspace.WithName(name), testspace.WithTierNameAndHashLabelFor(tier.NSTemplateTier), testspace.WithSpecTargetCluster(targetCluster.ClusterName))
 		spaces = append(spaces, s.Name)
+		t.Cleanup(func() {
+			cl := awaitilities.Host().Client
+			fresh := &toolchainv1alpha1.Space{}
+			require.NoError(t, cl.Get(context.TODO(), client.ObjectKeyFromObject(s), fresh))
+			require.NoError(t, cl.Delete(context.TODO(), fresh))
+		})
```

Comment:
that's weird - I went thought the functions that are being called and the Spaces are already added to the cleanup tasks: https://github.com/codeready-toolchain/toolchain-e2e/blob/eecfa1c48450277f333bff37b474cd7b5ac91cd3/testsupport/wait/host.go#L2632-L2633 So they should be deleted